File: flutter_nekokit/flutter_nekokit/android/src/main/kotlin/com/example/flutter_nekokit/FlutterNekokitPlugin.kt
package com.example.flutter_nekokit

import android.content.Context
import androidx.annotation.NonNull

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

// IMPORTANT: You will need to manually integrate the NekoBoxForAndroid's sing-box core (libcore)
// by compiling it into an AAR and adding it as a dependency in your build.gradle.
// The following imports are placeholders and assume the NativeInterface exists in the compiled AAR.
// Example: import moe.matsuri.nb4a.NativeInterface
// Example: import moe.matsuri.nb4a.SingBoxOptions

/** FlutterNekokitPlugin */
class FlutterNekokitPlugin: FlutterPlugin, MethodCallHandler {
  private lateinit var channel: MethodChannel
  private lateinit var applicationContext: Context

  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    applicationContext = flutterPluginBinding.applicationContext
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "flutter_nekokit")
    channel.setMethodCallHandler(this)
  }

  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
    when (call.method) {
      "initNekoBox" -> {
        // Placeholder: Call the actual NekoBox/sing-box initialization method here.
        // This assumes a method like `NativeInterface.init(applicationContext)` exists
        // in the integrated NekoBoxForAndroid core AAR.
        // NativeInterface.init(applicationContext)
        result.success(null)
      }
      "startProxy" -> {
        val config = call.argument<String>("config")
        // Placeholder: Call the actual NekoBox/sing-box start method with the config.
        // This assumes a method like `NativeInterface.start(config)` exists.
        // if (config != null) {
        //   NativeInterface.start(config)
        // }
        result.success(null)
      }
      "stopProxy" -> {
        // Placeholder: Call the actual NekoBox/sing-box stop method.
        // NativeInterface.stop()
        result.success(null)
      }
      "getProxyStatus" -> {
        // Placeholder: Call the actual NekoBox/sing-box status method.
        // result.success(NativeInterface.getStatus())
        result.success("Android: Proxy status not implemented (requires NekoBox core integration)")
      }
      "getConnectionStats" -> {
        // Placeholder: Call the actual NekoBox/sing-box connection stats method.
        // result.success(NativeInterface.getStats())
        result.success("Android: Connection stats not implemented (requires NekoBox core integration)")
      }
      "getVersion" -> {
        // Placeholder: Call the actual NekoBox/sing-box version method.
        // result.success(NativeInterface.getVersion())
        result.success("Android: Version not implemented (requires NekoBox core integration)")
      }
      "updateConfig" -> {
        val config = call.argument<String>("config")
        // Placeholder: Call the actual NekoBox/sing-box update config method.
        // if (config != null) {
        //   NativeInterface.updateConfig(config)
        // }
        result.success(null)
      }
      else -> {
        result.notImplemented()
      }
    }
  }

  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }
}

File: flutter_nekokit/flutter_nekokit/ios/Classes/FlutterNekokitPlugin.swift
import Flutter
import UIKit

// IMPORTANT: You will need to manually integrate the sing-box core for iOS.
// This typically involves compiling the Go sing-box library into an iOS framework
// and linking it to your Xcode project.
// The following methods are placeholders and assume the sing-box core provides
// equivalent functionalities to NekoBoxForAndroid.

public class FlutterNekokitPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "flutter_nekokit", binaryMessenger: registrar.messenger())
    let instance = FlutterNekokitPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "initNekoBox":
      // Placeholder: Call the actual sing-box initialization method here.
      // Example: SingBoxCore.shared.initialize()
      result(nil)
    case "startProxy":
      if let args = call.arguments as? [String: Any], let config = args["config"] as? String {
        // Placeholder: Call the actual sing-box start method with the config.
        // Example: SingBoxCore.shared.start(config: config)
        result(nil)
      } else {
        result(FlutterError(code: "INVALID_ARGUMENT", message: "Config argument missing or invalid", details: nil))
      }
    case "stopProxy":
      // Placeholder: Call the actual sing-box stop method.
      // Example: SingBoxCore.shared.stop()
      result(nil)
    case "getProxyStatus":
      // Placeholder: Call the actual sing-box status method.
      // Example: result(SingBoxCore.shared.getStatus())
      result("iOS: Proxy status not implemented (requires sing-box core integration)")
    case "getConnectionStats":
      // Placeholder: Call the actual sing-box connection stats method.
      // Example: result(SingBoxCore.shared.getStats())
      result("iOS: Connection stats not implemented (requires sing-box core integration)")
    case "getVersion":
      // Placeholder: Call the actual sing-box version method.
      // Example: result(SingBoxCore.shared.getVersion())
      result("iOS: Version not implemented (requires sing-box core integration)")
    case "updateConfig":
      if let args = call.arguments as? [String: Any], let config = args["config"] as? String {
        // Placeholder: Call the actual sing-box update config method.
        // Example: SingBoxCore.shared.updateConfig(config: config)
        result(nil)
      } else {
        result(FlutterError(code: "INVALID_ARGUMENT", message: "Config argument missing or invalid", details: nil))
      }
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}


